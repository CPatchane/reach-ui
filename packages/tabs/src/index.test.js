import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Tabs, TabList, Tab, TabPanels, TabPanel } from "./index";

const { click, keyDown } = fireEvent;

function getTabPanel(container, button) {
  return container.querySelector(`#${button.getAttribute("aria-controls")}`);
}

describe("rendering basic tabs", () => {
  beforeEach(() => {
    document.body.innerHTML = "";
    jest.resetModules();
  });

  it("should navigate with the mouse", () => {
    const { container, getByText, asFragment } = render(<BasicTabs />);
    const tab1 = getByText("Tab One");
    const tab2 = getByText("Tab Two");
    const tab3 = getByText("Tab Three");

    click(tab1);

    expect(getTabPanel(container, tab1)).toBeVisible();
    expect(getTabPanel(container, tab2)).not.toBeVisible();
    expect(getTabPanel(container, tab3)).not.toBeVisible();
    expect(asFragment()).toMatchSnapshot("tab 1 clicked");

    click(tab2);

    expect(getTabPanel(container, tab2)).toBeVisible();
    expect(getTabPanel(container, tab1)).not.toBeVisible();
    expect(getTabPanel(container, tab3)).not.toBeVisible();
    expect(asFragment()).toMatchSnapshot("tab 2 clicked");

    click(tab1);
    click(tab3);

    expect(getTabPanel(container, tab3)).toBeVisible();
    expect(getTabPanel(container, tab1)).not.toBeVisible();
    expect(getTabPanel(container, tab2)).not.toBeVisible();
    expect(asFragment()).toMatchSnapshot("tab 1 then 3 clicked");
  });

  it("should navigate with the keyboard", () => {
    const { container, getByText, getByRole, asFragment } = render(
      <BasicTabs />
    );
    const tab1 = getByText("Tab One");
    const tab2 = getByText("Tab Two");
    const tab3 = getByText("Tab Three");
    const tabList = getByRole("tablist");

    click(tab1);

    keyDown(tabList, { key: "ArrowRight", code: 39 });
    expect(document.activeElement).toBe(tab2);
    expect(getTabPanel(container, tab2)).toBeVisible();
    expect(getTabPanel(container, tab1)).not.toBeVisible();
    expect(asFragment()).toMatchSnapshot("keyed right to tab 2");

    keyDown(tabList, { key: "ArrowRight", code: 39 });
    expect(document.activeElement).toBe(tab3);
    expect(getTabPanel(container, tab3)).toBeVisible();
    expect(getTabPanel(container, tab2)).not.toBeVisible();
    expect(asFragment()).toMatchSnapshot("keyed right to tab 3");

    keyDown(tabList, { key: "ArrowRight", code: 39 });
    expect(document.activeElement).toBe(tab1);
    expect(getTabPanel(container, tab1)).toBeVisible();
    expect(getTabPanel(container, tab3)).not.toBeVisible();
    expect(asFragment()).toMatchSnapshot("keyed right back to the start");

    keyDown(tabList, { key: "ArrowRight", code: 39 });
    keyDown(tabList, { key: "ArrowRight", code: 39 });
    expect(document.activeElement).toBe(tab3);
    expect(getTabPanel(container, tab3)).toBeVisible();
    expect(getTabPanel(container, tab1)).not.toBeVisible();
    expect(asFragment()).toMatchSnapshot("keyed right twice to tab 3");

    keyDown(tabList, { key: "ArrowLeft", code: 37 });
    expect(document.activeElement).toBe(tab2);
    expect(getTabPanel(container, tab2)).toBeVisible();
    expect(getTabPanel(container, tab3)).not.toBeVisible();
    expect(asFragment()).toMatchSnapshot("keyed left to tab 2");

    keyDown(tabList, { key: "ArrowLeft", code: 37 });
    keyDown(tabList, { key: "ArrowLeft", code: 37 });
    expect(document.activeElement).toBe(tab3);
    expect(getTabPanel(container, tab3)).toBeVisible();
    expect(getTabPanel(container, tab2)).not.toBeVisible();
    expect(asFragment()).toMatchSnapshot("keyed left twice back to tab 3");

    keyDown(tabList, { key: "Home", code: 36 });
    expect(document.activeElement).toBe(tab1);
    expect(getTabPanel(container, tab1)).toBeVisible();
    expect(getTabPanel(container, tab3)).not.toBeVisible();
    expect(asFragment()).toMatchSnapshot("keyed home");

    keyDown(tabList, { key: "End", code: 35 });
    expect(document.activeElement).toBe(tab3);
    expect(getTabPanel(container, tab3)).toBeVisible();
    expect(getTabPanel(container, tab1)).not.toBeVisible();
    expect(asFragment()).toMatchSnapshot("keyed end");
  });
});

////////////////////////////////////////////////////////////////////////////////
function BasicTabs() {
  return (
    <div>
      <Tabs>
        <TabList>
          <Tab id="tab-1">Tab One</Tab>
          <Tab id="tab-2">Tab Two</Tab>
          <Tab id="tab-3">Tab Three</Tab>
        </TabList>

        <TabPanels>
          <TabPanel>
            <h1>one!</h1>
            <button>yo</button>
          </TabPanel>
          <TabPanel>
            <h1>two!</h1>
          </TabPanel>
          <TabPanel>
            <h1>three!</h1>
          </TabPanel>
        </TabPanels>
      </Tabs>
    </div>
  );
}
