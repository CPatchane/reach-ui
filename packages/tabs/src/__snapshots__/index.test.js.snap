// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering basic tabs should navigate with the keyboard: keyed end 1`] = `
<DocumentFragment>
  <div>
    <div
      data-reach-tabs=""
    >
      <div
        data-reach-tab-list=""
        role="tablist"
      >
        <button
          aria-controls="panel--_REACH_ID_--0"
          aria-selected="false"
          data-reach-tab=""
          id="tab-1"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab One
        </button>
        <button
          aria-controls="panel--_REACH_ID_--1"
          aria-selected="false"
          data-reach-tab=""
          id="tab-2"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab Two
        </button>
        <button
          aria-controls="panel--_REACH_ID_--2"
          aria-selected="true"
          data-reach-tab=""
          data-selected=""
          id="tab-3"
          role="tab"
          tabindex="0"
          type="button"
        >
          Tab Three
        </button>
      </div>
      <div
        data-reach-tab-panels=""
      >
        <div
          aria-labelledby="tab--_REACH_ID_--0"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--0"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            one!
          </h1>
          <button>
            yo
          </button>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--1"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--1"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            two!
          </h1>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--2"
          data-reach-tab-panel=""
          id="panel--_REACH_ID_--2"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            three!
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendering basic tabs should navigate with the keyboard: keyed home 1`] = `
<DocumentFragment>
  <div>
    <div
      data-reach-tabs=""
    >
      <div
        data-reach-tab-list=""
        role="tablist"
      >
        <button
          aria-controls="panel--_REACH_ID_--0"
          aria-selected="true"
          data-reach-tab=""
          data-selected=""
          id="tab-1"
          role="tab"
          tabindex="0"
          type="button"
        >
          Tab One
        </button>
        <button
          aria-controls="panel--_REACH_ID_--1"
          aria-selected="false"
          data-reach-tab=""
          id="tab-2"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab Two
        </button>
        <button
          aria-controls="panel--_REACH_ID_--2"
          aria-selected="false"
          data-reach-tab=""
          id="tab-3"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab Three
        </button>
      </div>
      <div
        data-reach-tab-panels=""
      >
        <div
          aria-labelledby="tab--_REACH_ID_--0"
          data-reach-tab-panel=""
          id="panel--_REACH_ID_--0"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            one!
          </h1>
          <button>
            yo
          </button>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--1"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--1"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            two!
          </h1>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--2"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--2"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            three!
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendering basic tabs should navigate with the keyboard: keyed left to tab 2 1`] = `
<DocumentFragment>
  <div>
    <div
      data-reach-tabs=""
    >
      <div
        data-reach-tab-list=""
        role="tablist"
      >
        <button
          aria-controls="panel--_REACH_ID_--0"
          aria-selected="false"
          data-reach-tab=""
          id="tab-1"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab One
        </button>
        <button
          aria-controls="panel--_REACH_ID_--1"
          aria-selected="true"
          data-reach-tab=""
          data-selected=""
          id="tab-2"
          role="tab"
          tabindex="0"
          type="button"
        >
          Tab Two
        </button>
        <button
          aria-controls="panel--_REACH_ID_--2"
          aria-selected="false"
          data-reach-tab=""
          id="tab-3"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab Three
        </button>
      </div>
      <div
        data-reach-tab-panels=""
      >
        <div
          aria-labelledby="tab--_REACH_ID_--0"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--0"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            one!
          </h1>
          <button>
            yo
          </button>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--1"
          data-reach-tab-panel=""
          id="panel--_REACH_ID_--1"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            two!
          </h1>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--2"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--2"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            three!
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendering basic tabs should navigate with the keyboard: keyed left twice back to tab 3 1`] = `
<DocumentFragment>
  <div>
    <div
      data-reach-tabs=""
    >
      <div
        data-reach-tab-list=""
        role="tablist"
      >
        <button
          aria-controls="panel--_REACH_ID_--0"
          aria-selected="false"
          data-reach-tab=""
          id="tab-1"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab One
        </button>
        <button
          aria-controls="panel--_REACH_ID_--1"
          aria-selected="false"
          data-reach-tab=""
          id="tab-2"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab Two
        </button>
        <button
          aria-controls="panel--_REACH_ID_--2"
          aria-selected="true"
          data-reach-tab=""
          data-selected=""
          id="tab-3"
          role="tab"
          tabindex="0"
          type="button"
        >
          Tab Three
        </button>
      </div>
      <div
        data-reach-tab-panels=""
      >
        <div
          aria-labelledby="tab--_REACH_ID_--0"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--0"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            one!
          </h1>
          <button>
            yo
          </button>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--1"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--1"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            two!
          </h1>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--2"
          data-reach-tab-panel=""
          id="panel--_REACH_ID_--2"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            three!
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendering basic tabs should navigate with the keyboard: keyed right back to the start 1`] = `
<DocumentFragment>
  <div>
    <div
      data-reach-tabs=""
    >
      <div
        data-reach-tab-list=""
        role="tablist"
      >
        <button
          aria-controls="panel--_REACH_ID_--0"
          aria-selected="true"
          data-reach-tab=""
          data-selected=""
          id="tab-1"
          role="tab"
          tabindex="0"
          type="button"
        >
          Tab One
        </button>
        <button
          aria-controls="panel--_REACH_ID_--1"
          aria-selected="false"
          data-reach-tab=""
          id="tab-2"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab Two
        </button>
        <button
          aria-controls="panel--_REACH_ID_--2"
          aria-selected="false"
          data-reach-tab=""
          id="tab-3"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab Three
        </button>
      </div>
      <div
        data-reach-tab-panels=""
      >
        <div
          aria-labelledby="tab--_REACH_ID_--0"
          data-reach-tab-panel=""
          id="panel--_REACH_ID_--0"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            one!
          </h1>
          <button>
            yo
          </button>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--1"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--1"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            two!
          </h1>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--2"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--2"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            three!
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendering basic tabs should navigate with the keyboard: keyed right to tab 2 1`] = `
<DocumentFragment>
  <div>
    <div
      data-reach-tabs=""
    >
      <div
        data-reach-tab-list=""
        role="tablist"
      >
        <button
          aria-controls="panel--_REACH_ID_--0"
          aria-selected="false"
          data-reach-tab=""
          id="tab-1"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab One
        </button>
        <button
          aria-controls="panel--_REACH_ID_--1"
          aria-selected="true"
          data-reach-tab=""
          data-selected=""
          id="tab-2"
          role="tab"
          tabindex="0"
          type="button"
        >
          Tab Two
        </button>
        <button
          aria-controls="panel--_REACH_ID_--2"
          aria-selected="false"
          data-reach-tab=""
          id="tab-3"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab Three
        </button>
      </div>
      <div
        data-reach-tab-panels=""
      >
        <div
          aria-labelledby="tab--_REACH_ID_--0"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--0"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            one!
          </h1>
          <button>
            yo
          </button>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--1"
          data-reach-tab-panel=""
          id="panel--_REACH_ID_--1"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            two!
          </h1>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--2"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--2"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            three!
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendering basic tabs should navigate with the keyboard: keyed right to tab 3 1`] = `
<DocumentFragment>
  <div>
    <div
      data-reach-tabs=""
    >
      <div
        data-reach-tab-list=""
        role="tablist"
      >
        <button
          aria-controls="panel--_REACH_ID_--0"
          aria-selected="false"
          data-reach-tab=""
          id="tab-1"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab One
        </button>
        <button
          aria-controls="panel--_REACH_ID_--1"
          aria-selected="false"
          data-reach-tab=""
          id="tab-2"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab Two
        </button>
        <button
          aria-controls="panel--_REACH_ID_--2"
          aria-selected="true"
          data-reach-tab=""
          data-selected=""
          id="tab-3"
          role="tab"
          tabindex="0"
          type="button"
        >
          Tab Three
        </button>
      </div>
      <div
        data-reach-tab-panels=""
      >
        <div
          aria-labelledby="tab--_REACH_ID_--0"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--0"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            one!
          </h1>
          <button>
            yo
          </button>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--1"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--1"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            two!
          </h1>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--2"
          data-reach-tab-panel=""
          id="panel--_REACH_ID_--2"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            three!
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendering basic tabs should navigate with the keyboard: keyed right twice to tab 3 1`] = `
<DocumentFragment>
  <div>
    <div
      data-reach-tabs=""
    >
      <div
        data-reach-tab-list=""
        role="tablist"
      >
        <button
          aria-controls="panel--_REACH_ID_--0"
          aria-selected="false"
          data-reach-tab=""
          id="tab-1"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab One
        </button>
        <button
          aria-controls="panel--_REACH_ID_--1"
          aria-selected="false"
          data-reach-tab=""
          id="tab-2"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab Two
        </button>
        <button
          aria-controls="panel--_REACH_ID_--2"
          aria-selected="true"
          data-reach-tab=""
          data-selected=""
          id="tab-3"
          role="tab"
          tabindex="0"
          type="button"
        >
          Tab Three
        </button>
      </div>
      <div
        data-reach-tab-panels=""
      >
        <div
          aria-labelledby="tab--_REACH_ID_--0"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--0"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            one!
          </h1>
          <button>
            yo
          </button>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--1"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--1"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            two!
          </h1>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--2"
          data-reach-tab-panel=""
          id="panel--_REACH_ID_--2"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            three!
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendering basic tabs should navigate with the mouse: tab 1 clicked 1`] = `
<DocumentFragment>
  <div>
    <div
      data-reach-tabs=""
    >
      <div
        data-reach-tab-list=""
        role="tablist"
      >
        <button
          aria-controls="panel--_REACH_ID_--0"
          aria-selected="true"
          data-reach-tab=""
          data-selected=""
          id="tab-1"
          role="tab"
          tabindex="0"
          type="button"
        >
          Tab One
        </button>
        <button
          aria-controls="panel--_REACH_ID_--1"
          aria-selected="false"
          data-reach-tab=""
          id="tab-2"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab Two
        </button>
        <button
          aria-controls="panel--_REACH_ID_--2"
          aria-selected="false"
          data-reach-tab=""
          id="tab-3"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab Three
        </button>
      </div>
      <div
        data-reach-tab-panels=""
      >
        <div
          aria-labelledby="tab--_REACH_ID_--0"
          data-reach-tab-panel=""
          id="panel--_REACH_ID_--0"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            one!
          </h1>
          <button>
            yo
          </button>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--1"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--1"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            two!
          </h1>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--2"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--2"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            three!
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendering basic tabs should navigate with the mouse: tab 1 then 3 clicked 1`] = `
<DocumentFragment>
  <div>
    <div
      data-reach-tabs=""
    >
      <div
        data-reach-tab-list=""
        role="tablist"
      >
        <button
          aria-controls="panel--_REACH_ID_--0"
          aria-selected="false"
          data-reach-tab=""
          id="tab-1"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab One
        </button>
        <button
          aria-controls="panel--_REACH_ID_--1"
          aria-selected="false"
          data-reach-tab=""
          id="tab-2"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab Two
        </button>
        <button
          aria-controls="panel--_REACH_ID_--2"
          aria-selected="true"
          data-reach-tab=""
          data-selected=""
          id="tab-3"
          role="tab"
          tabindex="0"
          type="button"
        >
          Tab Three
        </button>
      </div>
      <div
        data-reach-tab-panels=""
      >
        <div
          aria-labelledby="tab--_REACH_ID_--0"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--0"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            one!
          </h1>
          <button>
            yo
          </button>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--1"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--1"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            two!
          </h1>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--2"
          data-reach-tab-panel=""
          id="panel--_REACH_ID_--2"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            three!
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendering basic tabs should navigate with the mouse: tab 2 clicked 1`] = `
<DocumentFragment>
  <div>
    <div
      data-reach-tabs=""
    >
      <div
        data-reach-tab-list=""
        role="tablist"
      >
        <button
          aria-controls="panel--_REACH_ID_--0"
          aria-selected="false"
          data-reach-tab=""
          id="tab-1"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab One
        </button>
        <button
          aria-controls="panel--_REACH_ID_--1"
          aria-selected="true"
          data-reach-tab=""
          data-selected=""
          id="tab-2"
          role="tab"
          tabindex="0"
          type="button"
        >
          Tab Two
        </button>
        <button
          aria-controls="panel--_REACH_ID_--2"
          aria-selected="false"
          data-reach-tab=""
          id="tab-3"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab Three
        </button>
      </div>
      <div
        data-reach-tab-panels=""
      >
        <div
          aria-labelledby="tab--_REACH_ID_--0"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--0"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            one!
          </h1>
          <button>
            yo
          </button>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--1"
          data-reach-tab-panel=""
          id="panel--_REACH_ID_--1"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            two!
          </h1>
        </div>
        <div
          aria-labelledby="tab--_REACH_ID_--2"
          data-reach-tab-panel=""
          hidden=""
          id="panel--_REACH_ID_--2"
          role="tabpanel"
          tabindex="-1"
        >
          <h1>
            three!
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
